http_logging.go

```diff
@@ -74,6 +75,96 @@ func LogRequest(r *http.Request, msg string, extraAttributes ...KeyVal) {
 	S(Info, msg, attr...)
 }
 
+// LogResponse logs the response code, byte size and duration of the request.
+// additional key:value pairs can be passed as extraAttributes.
+//
+//nolint:revive // name is fine.
+func LogResponse[T *ResponseRecorder | *http.Response](r T, msg string, extraAttributes ...KeyVal) {
+	if !Log(Info) {
+		return
+	}
+	var status int
+	var size int64
+	switch v := any(r).(type) { // go generics...
+	case *ResponseRecorder:
+		status = v.StatusCode
+		size = v.ContentLength
+	case *http.Response:
+		status = v.StatusCode
+		size = v.ContentLength
+	}
+	attr := []KeyVal{
+		Int("status", status),
+		Int64("size", size),
+	}
+	attr = append(attr, extraAttributes...)
+	S(Info, msg, attr...)
+}
+
+// Can be used (and is used by LogAndCall()) to wrap a http.ResponseWriter to record status code and size.
+type ResponseRecorder struct {
+	w             http.ResponseWriter
+	startTime     time.Time
+	StatusCode    int
+	ContentLength int64
+}
+
+func (rr *ResponseRecorder) Header() http.Header {
+	return rr.w.Header()
+}
+
+func (rr *ResponseRecorder) Write(p []byte) (int, error) {
+	size, err := rr.w.Write(p)
+	rr.ContentLength += int64(size)
+	if err != nil {
+		rr.StatusCode = http.StatusInternalServerError
+	} else if rr.StatusCode == 0 {
+		rr.StatusCode = http.StatusOK
+	}
+	return size, err
+}
+
+func (rr *ResponseRecorder) WriteHeader(code int) {
+	rr.w.WriteHeader(code)
+	rr.StatusCode = code
+}
+
+// LogAndCall logs the incoming request and the response code, byte size and duration
+// of the request.
+//
+// If Config.CombineRequestAndResponse or the LOGGER_COMBINE_REQUEST_AND_RESPONSE
+// environment variable is true. then a single log entry is done combining request and
```

Comment:
```suggestion
// environment variable is true. Then a single log entry is done combining request and
```

Reply:
it's a typo I didn't see because my eyes are failing me after 14h, should have been a coma not a period, fixed